packages/react-components/src/components/table-output-component.tsx

```diff
@@ -212,23 +221,64 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         this.setState({
             tableColumns: this.columnArray
         });
+        if (this.columnApi) {
+            const columns = this.columnApi.getAllColumns();
+            const timestampHeader = columns.find(column => column.getColDef().headerName === 'Timestamp ns');
+            if (timestampHeader) {
+                this.timestampCol = timestampHeader.getColDef().field;
+            }
+        }
+
     }
 
     private async handleTimeSelectionChange(range: TimelineChart.TimeGraphRange) {
         const start = Math.trunc(this.props.range.getstart() + range.start);
-        if (this.lastIndex.timestamp === start) {
+        const end = Math.trunc(this.props.range.getstart() + range.end);
+
+        this.startTimeStamp = Math.min(start, end);
+        this.endTimeStamp = Math.max(start, end);
+        const startIndex = await this.fetchTableIndex(start);
+        const endIndex = await this.fetchTableIndex(end);
+
+        if (this.lastSelectedRange.startTimestamp === start &&
+            this.lastSelectedRange.endTimestamp === end &&
```

Comment:
In the old code, we only cared about start time. When you drag a selection, you receive many callbacks but only the end time changes, the start time is the same. The old code would return early.

In the new code, all those callbacks make you go to the trace-server twice, once for startIndex and once for endIndex. This can overload the server, which sometimes you can see the table takes a few seconds to be updated, and sometimes the table bounces around processing all those pending server responses coming back.

In addition, it doesn't seem that the endIndex is used at all, since you select rows by their timestamp.

If in a subsequent patch we make it select the endIndex when extending a selection, we should find a way to throttle that or make it happen when the selection is finished (mouse released).

Reply:
I agree we may not need the endIndex at all for now since I am using the timestamp for range selection. I removed it. Thanks for the suggestion.